java-agent/README.md

```diff
@@ -0,0 +1,69 @@
+# Apache SkyWalking Agent docker file
+
+**Docker images are not official ASF releases but provided for convenience. Recommended usage is always to build the
+source**
+
+<img src="http://skywalking.apache.org/assets/logo.svg" alt="Sky Walking logo" height="90px" align="right" />
+
+**SkyWalking**: an APM(application performance monitor) system, especially designed for microservices, cloud native and
+container-based (Docker, Kubernetes, Mesos) architectures.
+
+[![GitHub stars](https://img.shields.io/github/stars/apache/skywalking.svg?style=for-the-badge&label=Stars&logo=github)](https://github.com/apache/skywalking)
+[![Twitter Follow](https://img.shields.io/twitter/follow/asfskywalking.svg?style=for-the-badge&label=Follow&logo=twitter)](https://twitter.com/AsfSkyWalking)
+
+You could find docker file at [here](https://github.com/apache/skywalking-docker)
+
+This image only hosts the pre-built SkyWalking Java agent jars, and provides some convenient configurations for
+containerization scenarios.
+
+# How to use this image
+
+## Build your Java application image on top of this image
+
+```dockerfile
+FROM apache/skywalking-java-agent:8.5.0-jdk8
+
+# ... build your java application
+```
+
+You can start your Java application with `CMD` or `ENTRYPOINT`, but you don't need to care about the Java options to
+enable SkyWalking agent, it should be adopted automatically.
+
+## Use this image as sidecar of Kubernetes service
+
+In Kubernetes scenarios, you can also use this agent image as a sidecar.
```

Comment:
This is not the ultimate goal, the ultimate goal would be https://github.com/apache/skywalking/issues/6029 , auto injection of sidecar

Reply:
When he does this, please document what user should do on his application. Such as how to write `-javaagent:xxx`

Reply:
> When he does this, please document what user should do on his application. Such as how to write `-javaagent:xxx`

If the auto injection is done in SWCK, I think there is nothing for the users to do except for labeling the namespace such as `skywalking-agent-injection=true`, all the remaining operations are done via the SWCK operator @hanahmily right?

Reply:
I don't think SWCK will affect how user set their Java booting script. Do I miss anything?

Reply:
Take a look at https://github.com/apache/skywalking-docker/blob/74be4f870334af2cf0707c755240d54842e08195/java-agent/Dockerfile#L27 , and I believe our operator is able to set this env var after injecting the agent, no?

Reply:
Why a system env could make users don't need to modify? Could do be more clear?

They still to change like, don't they?
> java $JAVA_TOOL_OPTIONS xxx.jar

Reply:
> Why a system env could make users don't need to modify? Could do be more clear?
> 
> They still to change like, don't they?
> 
> > java $JAVA_TOOL_OPTIONS xxx.jar

No, JAVA_TOOL_OPTIONS is automatically adopted by JVM, this is the typical use of this option IMO. Check [this](https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/envvars002.html)

> In many environments the command line is not readily accessible to start the application with necessary command-line options. This often arises with applications that use embedded VMs (meaning they use the Java Native Interface (JNI) Invocation API to start the VM), or where the startup is deeply nested in scripts. In these environments the JAVA_TOOL_OPTIONS environment variable can be useful to augment a command line.